packages/backend/routes/batches.js

```diff
@@ -138,6 +140,10 @@ router.patch("/update", withRole("admin"), async (req, res) => {
 
   if (batchContractAddress) {
     batchData.contractAddress = batchContractAddress;
+    batchData.nftContractAddress = await getNFTContractAddress(batchContractAddress);
```

Comment:
A bit of a nitpick:

This is being called every time you edit a batch, right? It doesn't matter if the batchContractAddress is the same. Since we already have `batch` loaded, it'd be easy to check if they are the same or not.

Not a big deal ofc.. but in case you want to fix this little inefficiency. 

You also said:

> (There is no migration. We can just edit the batches and submit it to the database, then the data should be fetched.)

This won't work if we make this change tho. We'd need to create a migration script.

Up to you!

Reply:
Sorry, I overlooked it, will do it tomorrow.